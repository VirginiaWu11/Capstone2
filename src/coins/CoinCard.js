import React, { memo, useCallback } from "react";
import Card from "@mui/material/Card";
import CardContent from "@mui/material/CardContent";
import Typography from "@mui/material/Typography";
import Box from "@mui/material/Box";
import CardMedia from "@mui/material/CardMedia";
import IconButton from "@mui/material/IconButton";
import Tooltip from "@mui/material/Tooltip";

import PushPinIcon from "@mui/icons-material/PushPin";
import PushPinOutlinedIcon from "@mui/icons-material/PushPinOutlined";

const CoinCard = memo(({ coin, isPinned, handlePin, handleUnpin }) => {
  const truncate = useCallback((str, n) => {
    return str.length > n ? str.substr(0, n - 1) + "..." : str;
  }, []);

  return (
    <Card
      sx={{ minWidth: 275, display: "flex", justifyContent: "space-between" }}
    >
      <Box
        sx={{
          display: "flex",
          flexDirection: "column",
        }}
      >
        <CardContent>
          <Typography variant="h5" component="div">
            {truncate(coin.name, 15)}
          </Typography>
          <Typography sx={{ mb: 1.5 }} color="text.secondary">
            {truncate(coin.symbol, 20)}
          </Typography>
          {coin.current_price ? (
            <Typography variant="body2">{`Current Price: $${coin.current_price.toLocaleString()}`}</Typography>
          ) : (
            <Typography variant="body2">{`Current Price: Not Available`}</Typography>
          )}
          {coin.market_cap ? (
            <Typography variant="body2">{`Market Cap: $${coin.market_cap.toLocaleString()}`}</Typography>
          ) : (
            <Typography variant="body2">{`Market Cap: Not Available`}</Typography>
          )}
        </CardContent>
      </Box>
      <Box>
        {isPinned(coin.id) ? (
          <IconButton
            onMouseDown={(event) => event.stopPropagation()}
            onClick={(event) => {
              event.stopPropagation();
              event.preventDefault();
              console.log("IconButton clicked");
              handleUnpin(coin.id);
            }}
          >
            <Tooltip title="Unpin from Watchlist">
              <PushPinIcon color="primary" />
            </Tooltip>
          </IconButton>
        ) : (
          <IconButton
            onMouseDown={(event) => event.stopPropagation()}
            onClick={(event) => {
              event.stopPropagation();
              event.preventDefault();
              console.log("IconButton clicked");
              handlePin(coin.id);
            }}
          >
            <Tooltip title="Pin to Watchlist">
              <PushPinOutlinedIcon sx={{ transform: "rotate(0.20turn)" }} />
            </Tooltip>
          </IconButton>
        )}
        <IconButton
          onMouseDown={(event) => event.stopPropagation()}
          onClick={(event) => {
            event.stopPropagation();
            event.preventDefault();
            console.log("IconButton clicked");
          }}
        >
          me
        </IconButton>

        {coin.image && (
          <CardMedia
            component="img"
            sx={{
              objectFit: "contain",
              width: "5em",
              display: { sm: "block" },
              mr: 2,
            }}
            image={coin.image}
            alt={`${coin.name} logo`}
          />
        )}
      </Box>
    </Card>
  );
});
export default CoinCard;
