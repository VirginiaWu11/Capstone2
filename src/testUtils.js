import React from "react";
import { UserContext } from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
};

const UserProvider = ({
  children,
  currentUser = demoUser,
  setCurrentUser,
  token,
  setToken,
  signup,
  signin,
  signout,
  updateProfile,
  infoLoaded,
  setInfoLoaded,
}) => (
  <UserContext.Provider
    value={{
      currentUser,
      setCurrentUser,
      token,
      setToken,
      signup,
      signin,
      signout,
      updateProfile,
      infoLoaded,
      setInfoLoaded,
    }}
  >
    {children}
  </UserContext.Provider>
);

export { UserProvider };
