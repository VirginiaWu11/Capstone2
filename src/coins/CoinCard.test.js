import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import CoinCard from "./CoinCard";
import { UserProvider } from "../testUtils";

it("matches snapshot", function () {
  let coin = {
    id: "testcoin",
    name: "testcoin",
    symbol: "TC",
    current_price: 1,
    market_cap: 10,
  };

  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <CoinCard
          coin={coin}
          isPinned={() => false}
          handlePin={() => false}
          handleUnpin={() => false}
          isOnPortfolio={() => false}
          removeFromPortfolio={() => false}
          handlePortfolioModalOpen={() => false}
        />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
