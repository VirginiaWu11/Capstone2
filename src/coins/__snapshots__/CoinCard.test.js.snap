// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-165pmne-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiBox-root css-j7qwjs"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          testcoin
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1 css-9eg9l8-MuiTypography-root"
        >
          TC
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          Current Price: $1
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          Market Cap: $10
        </p>
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          aria-label="Pin to Watchlist"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1x8ha6j-MuiSvgIcon-root"
          data-mui-internal-clone-element="true"
          data-testid="PushPinOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14 4v5c0 1.12.37 2.16 1 3H9c.65-.86 1-1.9 1-3V4h4m3-2H7c-.55 0-1 .45-1 1s.45 1 1 1h1v5c0 1.66-1.34 3-3 3v2h5.97v7l1 1 1-1v-7H19v-2c-1.66 0-3-1.34-3-3V4h1c.55 0 1-.45 1-1s-.45-1-1-1z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          aria-label="Add to Portfolio"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-mui-internal-clone-element="true"
          data-testid="AccountBalanceWalletIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21 18v1c0 1.1-.9 2-2 2H5c-1.11 0-2-.9-2-2V5c0-1.1.89-2 2-2h14c1.1 0 2 .9 2 2v1h-9c-1.11 0-2 .9-2 2v8c0 1.1.89 2 2 2h9zm-9-2h10V8H12v8zm4-2.5c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
